core/src/main/java/org/nervos/ckb/service/adapter/IntegerTypeAdapter.java

```diff
@@ -25,11 +18,6 @@ public Integer deserialize(JsonElement json, Type typeOfT, JsonDeserializationCo
       return json.getAsInt();
     }
     String hexValue = json.getAsString().substring(2);
-    // deserialize 0xffffffff to -1 for outpoint index in coinbase transaction
-    if (Objects.equals("ffffffff", hexValue)) {
-      return -1;
-    } else {
-      return Integer.valueOf(hexValue, 16);
-    }
+    return Integer.parseUnsignedInt(hexValue, 16);
```

Comment:
Use [parseUnsignedLong](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Long.html#parseUnsignedLong(java.lang.CharSequence,int,int,int)) instead?

Reply:
No. This method returns a Integer.